src/core/HY_Features.cpp

```diff
@@ -3,67 +3,69 @@
 
 using namespace hy_features;
 
-HY_Features::HY_Features(geojson::GeoJSON linked_hydro_fabric, std::shared_ptr<Formulation_Manager> formulations)
-  :network(linked_hydro_fabric), formulations(formulations), fabric(linked_hydro_fabric)
+HY_Features::HY_Features( geojson::GeoJSON fabric, std::shared_ptr<Formulation_Manager> formulations)
+  :HY_Features(network::Network(fabric), formulations, fabric)
 {
-  init();
+  /* If you nest the constuctor calls like this, then the network, while valid in the next constructor,
+     is empty once the collection is fully instansiated...
+    HY_Features(network::Network(fabric));
+  */
 }
 
-HY_Features::HY_Features( geojson::GeoJSON catchments, std::string* link_key, std::shared_ptr<Formulation_Manager> formulations)
-    : network(catchments, link_key), formulations(formulations), fabric(catchments)
-{
-  init();
+HY_Features::HY_Features( geojson::GeoJSON catchments, std::string* link_key, std::shared_ptr<Formulation_Manager> formulations):
+    HY_Features(network::Network(catchments, link_key), formulations, catchments){  
 }
 
-void HY_Features::init()
+HY_Features::HY_Features(network::Network network, std::shared_ptr<Formulation_Manager> formulations, geojson::GeoJSON fabric)
+  :network(network), formulations(formulations)
 {
-  std::string feat_id;
-  std::string feat_type;
-  std::vector<std::string> origins, destinations;
+      std::string feat_id;
+      std::string feat_type;
+      std::vector<std::string> origins, destinations;
 
-  for(const auto& feat_idx : network)
-  {
-    feat_id = network.get_id(feat_idx);//feature->get_id();
-    feat_type = feat_id.substr(0, 3);
+      for(const auto& feat_idx : network){
+        feat_id = network.get_id(feat_idx);//feature->get_id();
+        feat_type = feat_id.substr(0, feat_id.find(hy_features::identifiers::seperator) );
 
-    destinations  = network.get_destination_ids(feat_id);
-    if(feat_type == "cat" || feat_type == "agg")
```

Comment:
I just noticed - this same change (and the one below in `else if`) should have been made for HY_Features_MPI as well.

Reply:
Fix in #674